presto-main/src/main/java/com/facebook/presto/operator/PartitionedLookupSourceFactory.java

```diff
@@ -248,24 +247,24 @@ private void checkAllFuturesDone()
     }
 
     @Override
-    public CompletableFuture<LookupSource> readSpilledLookupSource(Session session, int partition)
+    public synchronized CompletableFuture<LookupSource> readSpilledLookupSource(Session session, int partition)
```

Comment:
I see this is removed in next commit, so it's totally up to you what you do with my comment :)